api/src/test/java/org/openmrs/module/registrationapp/converter/RegistrationSummaryExtensionsGeneratorTest.java

```diff
@@ -0,0 +1,129 @@
+    	// setup
+    	InputStream inputStream = getClass().getClassLoader().getResourceAsStream("registration_app.json");    	
+    	List<AppDescriptor> appDescriptors = new ObjectMapper().readValue(inputStream, new TypeReference<List<AppDescriptor>>() {});
+    	
+    	AppDescriptor appDescriptor = appDescriptors.get(0);
+    	JsonNode sections = appDescriptor.getConfig().get("sections");
+    	
+    	// replay
+    	List<Extension> extensions = RegistrationSummaryExtensionsGenerator.generate(appDescriptor, false);
+    	
+    	// verify
+    	assertNotNull(extensions);
+    	assertEquals(3, extensions.size());
+    	for (Extension extn : extensions) {
+    		//For each Extension, loop over all appDescriptor's sections
+	    	for (JsonNode section : sections) {
+	    		String sectionId = section.get("id").getTextValue();
+	    		
+	    		if (extn.getId().contains(sectionId)) {
+					assertNotNull(extn.getId());
+					assertEquals("referenceapplication.registrationapp.summary." + sectionId, extn.getId());
+					
+					assertNotNull(extn.getAppId());
+					assertEquals(appDescriptor.getId(), extn.getAppId());
+					
+					assertNotNull(extn.getExtensionPointId());
+					assertEquals("registrationSummary.contentFragments", extn.getExtensionPointId());
+					
+					assertNotNull(extn.getExtensionParams());
+					assertEquals("summary/section", extn.getExtensionParams().get("fragment"));
+					assertEquals("registrationapp", extn.getExtensionParams().get("provider"));
+					
+					Map<String, String> fragmentConfig = (HashMap<String, String>) extn.getExtensionParams().get("fragmentConfig");
+					assertEquals(sectionId, fragmentConfig.get("sectionId"));
+					break;
+				}
+			}
+    	}
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void generate_shouldThrowWhenNotRegAppConfig() throws Exception {
+    	// setup
+    	AppDescriptor appDescriptor = new AppDescriptor("fooId", "fooDesc", null, null, null, null, 0);
+    	appDescriptor.setInstanceOf("not.a.reg.app");
+    	
+    	// replay
+    	RegistrationSummaryExtensionsGenerator.generate(appDescriptor, true);
+    }
+    
+    @Test
+    public void generate_shouldReturnEmptyWhenEmptySectionsInConfig() throws Exception {
+    	// setup
+    	AppDescriptor appDesc = new AppDescriptor("my.registrationapp.registerPatient", "Create a new Patient Record", "Register Patient", null, null, null, 0, null, null);
+    	appDesc.setInstanceOf("registrationapp.registerPatient");
+    	
+    	ObjectNode config = new ObjectMapper().createObjectNode();
+    	config.putArray("sections"); 
+    	appDesc.setConfig(config);
+    	
+    	// replay
+    	List<Extension> regSummaryExtensions = RegistrationSummaryExtensionsGenerator.generate(appDesc, true);
+
+    	// verify
+    	assertThat(regSummaryExtensions, is(empty()));
+    }
+    
+    @Test
+    public void extractExtensions_shouldExtractColumnWidgets() throws Exception {
```

Comment:
This test might go away if you provide a new test for the controller as explained in my headline comment.

Reply:
Correct.